### PR TITLE
Cccp 4534 Support additional test and production branches

### PR DESCRIPTION
**Pull request by khappone**
```
To make reviews easier, there is the code deobfuscated. The two later for loops were added to support more test and prod branches.

for env in $(ls {{puppet_code_dir}}/environments | awk ''$1 !~ /^master$|^test.*$|^production.*$|^development.*$/''); do
   ln -rs {{puppet_code_dir}}/environments/$env/hieradata/development.yaml {{puppet_code_dir}}/environments/$env/hieradata/$env.yaml;
   ln -rs {{puppet_code_dir}}/environments/$env/hieradata/secret/development.eyaml {{puppet_code_dir}}/environments/$env/hieradata/secret/$env.eyaml;
done;

for env in $(ls {{puppet_code_dir}}/environments | awk ''$1 ~ /^test_/''); do
   ln -rs {{puppet_code_dir}}/environments/$env/hieradata/test.yaml {{puppet_code_dir}}/environments/$env/hieradata/$env.yaml;
   ln -rs {{puppet_code_dir}}/environments/$env/hieradata/secret/test.eyaml {{puppet_code_dir}}/environments/$env/hieradata/secret/$env.eyaml;
done;

for env in $(ls {{puppet_code_dir}}/environments | awk ''$1 ~ /^production_/''); do
   ln -rs {{puppet_code_dir}}/environments/$env/hieradata/production.yaml {{puppet_code_dir}}/environments/$env/hieradata/$env.yaml;
   ln -rs {{puppet_code_dir}}/environments/$env/hieradata/secret/production.eyaml {{puppet_code_dir}}/environments/$env/hieradata/secret/$env.eyaml;
done; 

ln -rfs {{puppet_code_dir}}/environments/test {{puppet_code_dir}}/environments/development
```

### CODE REVIEW

File: templates/r10k.yaml [hunk 1]
```diff
@@ -5,5 +5,5 @@ sources:
...
...$(ls {{puppet_code_dir}}/environments | awk ''$1 !~ /^master$|^test.*$|^production.*$|^development.*$/''); do ln -rs {{puppet_code_dir}}/environments/$env/hieradata/development.yaml {{puppet_code_dir}}/environments/$env/hieradata/$env.yaml; ln -rs {{puppet_code_dir}}/environments/$env/hieradata/secret/development.eyaml {{puppet_code_dir}}/environments/$env/hieradata/secret/$env.eyaml; done; for env in $(ls {{puppet_code_dir}}/environments | awk ''$1 ~ /^test_/''); do ln -rs {{puppet_code_dir}}/environments/$env/hieradata/test.yaml {{puppet_code_dir}}/environments/$env/hieradata/$env.yaml; ln -rs {{puppet_code_dir}}/environments/$env/hieradata/secret/test.eyaml {{puppet_code_dir}}/environments/$env/hieradata/secret/$env.eyaml; done; for env in $(ls {{puppet_code_dir}}/environments | awk ''$1 ~ /^production_/''); do ln -rs {{puppet_code_dir}}/environments/$env/hieradata/production.yaml {{puppet_code_dir}}/environments/$env/hieradata/$env.yaml; ln -rs {{puppet_code_dir}}/environments/$env/hieradata/secret/production.eyaml {{puppet_code_dir}}/environments/$env/hieradata/secret/$env.eyaml; done; ln -rfs {{puppet_code_dir}}/environments/test {{puppet_code_dir}}/environments/development']
```
Comment (jpcsc) [inline templates/r10k.yaml:8]: I find the awk solution a bit confusing. Would you consider changing it to something like?

`for env in {{puppet_code_dir}}/environments/*/; do env=$(basename "$env"); if [[ $env == test_* ]]; then X; elif [[ $env == production_* ]]; then Y; else Z; fi; done`

Comment (khappone) [reply templates/r10k.yaml:8]: Based on yesterday's discussions we could do this. I think bash preferences depends on people though and we don't have to touch this code very often (third time ever now).

So I would prefer to go with the awk version, so we don't have to debug the another way, as this probably should be merged before tomorrow's placement.

**Review by jpcsc — APPROVED** [review]
looks alright

File: templates/r10k.yaml (hunk 1 above)
Comment (jpcsc) [reply templates/r10k.yaml:8]: Thats fine. As additional I think the old and the new code break if there would be file/dir with spaces. But anyway the improvement does not seem introduce more problems.

```
$ ls ./environments/
 11111111111   aismdasod   asdatestasdasd   development   master   master_asdasd   production_asdasd  'something with space'   test   test_123123   testasdasd
$ for env in $(ls ./environments | awk '$1 !~ /^master$|^test.*$|^production.*$|^development.*$/'); do echo "moi: $env"; done;
moi: 11111111111
moi: aismdasod
moi: asdatestasdasd
moi: master_asdasd
moi: something
moi: with
moi: space
```

**Pull request merged**